base_name_search_improved/models/ir_model.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2016 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, tools
+
+# Extended name search is only used on some operators
+ALLOWED_OPS = set(['ilike', 'like'])
+
+
+@tools.ormcache(skiparg=0)
+def _get_rec_names(self):
+    "List of fields to search into"
+    model = self.env['ir.model'].search(
+        [('model', '=', str(self._name))])
```

Comment:
What is the reason to use `str()` here?

Reply:
Hi  @moylop260 

Great! happy to read from you :)
Excellent review, thanks for taking the time in this PR,
I have applied the changes you suggested

Best Regards